syncer/syncer.go

```diff
@@ -716,6 +729,43 @@ func (s *Syncer) addCount(isFinished bool, queueBucket string, tp opType, n int6
 	}
 }
 
+func (s *Syncer) jobsIsEmpty() bool {
+	for idx := range s.jobs {
+		if len(s.jobs[idx]) > 0 {
+			return false
+		}
+	}
+	return true
+}
+
+// updateReplicationLag calc syncer replication lag by job, it called after every batch dml job / one skip job / one ddl
+// job is flushed to target db.
+func (s *Syncer) updateReplicationLag(job *job, queueBucketName string) {
+	if job != nil {
+		var lag int64
+		switch job.tp {
+		case skip, ddl:
+			lag = time.Now().Unix() + s.tsOffset.Load() - int64(job.eventHeader.Timestamp)
```

Comment:
Still a bit little strange here. For DDL it's reasonable here. Because DDL job will clean all the dmls before execution.
But a skip job won't. So the lag will be much smaller when we add this skip job but become larger after we finish a dml job. It's not supposed to happen.

Reply:
@lichunzhu 

fixed, and add add it test scripts, PTAL